src/context/auth-context.js

```diff
@@ -0,0 +1,94 @@
+import { useContext, createContext, useReducer } from "react";
+import axios from "axios";
+import { userInitialState, authReducer } from "reducer/auth-reducer";
+
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const [authState, authDispatch] = useReducer(authReducer, userInitialState);
+
+  //  While login
+  const loginHandler = async (e, formData) => {
+    try {
+      const response = await axios.post(
+        "/api/auth/login",
+        e.target.value === "user"
+          ? JSON.stringify(formData)
+          : {
+              email: "kajalkumari@gmail.com",
+              password: "kajalkumari",
+            }
+      );
+      //saving the token in localstorage
+      localStorage.setItem("token", response.data.encodedToken);
+      console.log(response.data.foundUser);
```

Comment:
remove console.log

Reply:
Will remove it after implementing cart wishlist functionality